sencha-workspace/SlateTasksStudent/app/view/AppHeader.js

```diff
@@ -19,6 +19,42 @@ Ext.define('SlateTasksStudent.view.AppHeader', {
             align: 'center'
         },
         items: [{
+            xtype: 'combo',
+            itemId: 'student-selector',
+            name: 'StudentSelector',
+            fieldLabel: 'Student',
+            valueField: 'Username',
+            allowBlank: true,
+            queryParam: 'q',
+            margin: '0 10 0 0',
+            displayTpl: Ext.create('Ext.XTemplate',
```

Comment:
@jtwiest no classes should ever be instantiated during class-definition-time. This expression, `Ext.create(...)`, will get executed while this source file is being parsed, bypassing the framework's loader's ability to dynamically resolve dependencies and sequence loading, creating a race condition. 

The correct why to do this would be to define `displayTpl` as an array:
```
displayTpl: [
    '<tpl for=".">'
       '...'
    '</tpl>'
]
```

Components that have `*Tpl` configs use the [lookupTpl](http://docs.sencha.com/extjs/6.2.0/modern/Ext.Widget.html#method-lookupTpl) method to read them which handles instantiating a string/array into a `Ext.XTemplate` instance as needed.

But also, a better solution here is to add `Slate.model.person.Person` to the `requires` for this class at the top, and then set `model: 'Slate.model.person.Person'` for the store instead of `fields`, and then just use `displayField: 'FullName'` which utilized the `FullName` virtual field provided by the official [Person model](https://github.com/SlateFoundation/slate-core-data/blob/master/src/model/person/Person.js)

Reply:
@themightychris Interesting, I pulled that right off the docs but I suppose that was in a different context ie the combo was bing created vie Ext.create no the method we're using.

http://docs.sencha.com/extjs/6.0.2/classic/Ext.form.field.ComboBox.html
(Customized combobox example)

What you're suggesting makes sense though, thanks for the explanation.